open_spiel/games/chess/chess_board.cc

```diff
@@ -451,30 +454,36 @@ Square ChessBoard<kBoardSize>::find(const Piece &piece) const {
   return InvalidSquare();
 }
 
-template <uint32_t kBoardSize>
-void ChessBoard<kBoardSize>::GenerateLegalMoves(
+template <uint32_t kBoardSize, bool kAllowCheck>
+void ChessBoard<kBoardSize, kAllowCheck>::GenerateLegalMoves(
     const MoveYieldFn &yield) const {
-  auto king_square = find(Piece{to_play_, PieceType::kKing});
 
-  GeneratePseudoLegalMoves([this, &king_square, &yield](const Move &move) {
-    // See if the move is legal by applying, checking whether the king is
-    // under attack, and undoing the move.
-    // TODO: Optimize this.
-    auto board_copy = *this;
-    board_copy.ApplyMove(move);
+  if (kAllowCheck) {
```

Comment:
Add a small comment, something like
// We do not need to filter moves that would result for King to move / stay in check, so we can yield all pseudo legal moves.